File: src/main/java/org/hwabeag/playertime/expansions/PlayerTimeExpansion.java
package org.hwabeag.playertime.expansions;

import org.bukkit.configuration.file.FileConfiguration;
import org.hwabeag.playertime.PlayerTime;
import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.OfflinePlayer;
import org.hwabeag.playertime.config.ConfigManager;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class PlayerTimeExpansion extends PlaceholderExpansion {


    FileConfiguration PlayerTimeConfig = ConfigManager.getConfig("playertime");
    private final PlayerTime plugin; //

    public PlayerTimeExpansion(PlayerTime plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean persist() {
        return true;
    }

    @Override
    public boolean canRegister() {
        return true;
    }

    @Override
    public @NotNull String getAuthor() {
        return plugin.getDescription().getAuthors().toString();
    }

    @Override
    public @NotNull String getIdentifier() {
        return "playertime";
    }

    @Override
    public @NotNull String getVersion() {
        return plugin.getDescription().getVersion();
    }
    @Override
    public String onRequest(OfflinePlayer player, @NotNull String params) {
        if (params.equalsIgnoreCase("get")) { // %playertime.player.get%
            String Name = Objects.requireNonNull(player).getName();
            if (PlayerTimeConfig.getString("playertime." + Name) == null) {
                return null;
            } else {
                return PlayerTimeConfig.getString("playertime." + Name);
            }
        }
        return null; //
    }
}

File: src/main/kotlin/org/hwabeag/playertime/expansions/PlayerTimeExpansion.kt
package org.hwabeag.playertime.expansions

import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.OfflinePlayer
import org.hwabeag.playertime.PlayerTime

class PlayerTimeExpansion(private val plugin: PlayerTime) : PlaceholderExpansion() {

    override fun persist(): Boolean = true

    override fun canRegister(): Boolean = true

    override fun getAuthor(): String = plugin.description.authors.joinToString(",")

    override fun getIdentifier(): String = "playertime"

    override fun getVersion(): String = plugin.description.version

    override fun onRequest(player: OfflinePlayer?, params: String): String? {
        val name = player?.name ?: return null
        return when (params.lowercase()) {
            "get", "player_get" -> (plugin.storage.getMinutesByName(name) ?: 0).toString()
            else -> null
        }
    }
}
